combine_fold_results.py

```diff
@@ -0,0 +1,235 @@
+    second data source of a combined data model. If test_rmsds is not
+    truthy, the rmsd and affinity args are ignored.'''
+    two = '2' if is_data2 else ''
+
+    #average, min, max test AUC for last 1000 iterations
+    last_iters = 1000
+    avg_auc, max_auc, min_auc = last_iters_statistics(test_aucs, test_interval, last_iters)
+    txt = 'For the last %s iterations:\nmean AUC=%.2f  max AUC=%.2f  min AUC=%.2f' % (last_iters, avg_auc, max_auc, min_auc)
+
+    #average aucs across folds
+    mean_test_aucs = np.mean(test_aucs, axis=0)
+    mean_train_aucs = np.mean(train_aucs, axis=0)
+
+    #write test and train aucs (mean and for each fold)
+    write_results_file('%s.auc.test%s' % (outprefix, two), mean_test_aucs, *test_aucs)
+    write_results_file('%s.auc.train%s' % (outprefix, two), mean_train_aucs, *train_aucs)
+
+    #training plot of mean auc across folds
+    training_plot('%s_auc_train%s.pdf' % (outprefix, two), mean_train_aucs, mean_test_aucs)
+
+    #roc curve for the last iteration - combine all tests
+    if len(np.unique(all_y_true)) > 1:
+        fpr, tpr, _ = sklearn.metrics.roc_curve(all_y_true, all_y_score)
+        auc = sklearn.metrics.roc_auc_score(all_y_true, all_y_score)
+        write_results_file('%s.auc.finaltest%s' % (outprefix, two), all_y_true, all_y_score, footer='AUC %f\n' % auc)
+        plot_roc_curve('%s_roc%s.pdf' % (outprefix, two), fpr, tpr, auc, txt)
+
+    if test_rmsds:
+
+        #average rmsds across folds
+        mean_test_rmsds = np.mean(test_rmsds, axis=0)
+        mean_train_rmsds = np.mean(train_rmsds, axis=0)
+
+        #write test and train rmsds (mean and for each fold)
+        write_results_file('%s.rmsd.test%s' % (outprefix, two), mean_test_rmsds, *test_rmsds)
+        write_results_file('%s.rmsd.train%s' % (outprefix, two), mean_train_rmsds, *train_rmsds)
+
+        #training plot of mean rmsd across folds
+        training_plot('%s_rmsd_train%s.pdf' % (outprefix, two), mean_train_rmsds, mean_test_rmsds)
+
+        all_y_aff = np.array(all_y_aff)
+        all_y_predaff = np.array(all_y_predaff)
+        yt = np.array(all_y_true, dtype=np.bool)
+        rmsdt = sklearn.metrics.mean_squared_error(all_y_aff[yt], all_y_predaff[yt])
+        r2t = sklearn.metrics.r2_score(all_y_aff[yt], all_y_predaff[yt])
+        write_results_file('%s.rmsd.finaltest%s' % (outprefix, two), all_y_aff, all_y_predaff, footer='RMSD,R^2 %f %f\n' % (rmsdt, r2t))
+
+        plot_correlation('%s_rmsd%s.pdf' % (outprefix, two), all_y_aff[yt], all_y_predaff[yt], rmsdt, r2t)
+
+
+def parse_args(argv=None):
+    parser = argparse.ArgumentParser(description='Combine training results from different folds and make graphs')
+    parser.add_argument('-o','--outprefix',type=str,required=True,help="Prefix for input and output files (--outprefix from train.py)")
+    parser.add_argument('-n','--numfolds',type=int,required=False,help="Number of folds to combine, default is 3",default='3')
```

Comment:
I would prefer if numfolds is not specified that the files are identified by globing for the appropriate pattern.  This seems more robust.